### PR TITLE
Refactor Outputstream to update buffer from select methods

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
The performance is unchanged with this PR. I would prefer to have only two methods (if even that) but I can't figure out a way without hitting some performance. Least hit I discovered was 12% if 
```swift
public func write(_ byte: UInt8) {
    buffer.append(byte)
}
```
becomes:
```swift
public func write(_ byte: UInt8) {
    var byte = byte
    let ptr = UnsafeBufferPointer(start: &byte, count: 1)
    write(ptr)
}
```

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
| Test | Old (sec) | Current PR (sec) |
| --- | :---: | :---: |
|test1MBOf16ByteArrays_X100 			         |0.308|  0.301
|test1MBOf16ByteArraySlice_X100  		         |0.317| 	0.310
|test1MBOf16ByteStrings_X10 			         |0.103|	0.103
|test1MBOf1KByteArrays_X1000 			         |0.214|	0.207
|test1MBOf1KByteStrings_X100 			         |0.581|	0.567
|test1MBOfByte_X10 						 |0.236|	0.231
|test1MBOfCharacters_X1					 |0.404|	0.406
|test1MBOfJSONEncoded16ByteStrings_X10 	 |0.098|	0.096
|testFormattedJSONOutput 				 |0.197|	0.187

File: Sources/Basic/OutputByteStream.swift [hunk 1]
```diff
@@ -73,20 +73,25 @@ public class OutputByteStream: OutputStream {
...
+
+    /// Write a sequence of bytes to the buffer.
+    public func write(_ bytes: [UInt8]) {
+        write(ArraySlice(bytes))
+    }
     
     /// Write a sequence of bytes to the buffer.
     public func write<S: Sequence where S.Iterator.Element == UInt8>(_ sequence: S) {
-        buffer += sequence
+        write(ArraySlice(sequence))
```
Comment (ddunbar) [inline Sources/Basic/OutputByteStream.swift]: This doesn't make a copy, right?

**Comment by ddunbar** [comment]
LGTM

File: Sources/Basic/OutputByteStream.swift (hunk 1 above)
Comment (aciidb0mb3r) [inline Sources/Basic/OutputByteStream.swift]: Not too sure... I'll try to find out

**Pull request merged**